runtime/interpreter/value.go

```diff
@@ -17735,13 +17792,18 @@ func (v *DictionaryValue) Clone(interpreter *Interpreter) Value {
 		panic(errors.NewExternalError(err))
 	}
 
-	return &DictionaryValue{
-		Type:             v.Type,
-		semaType:         v.semaType,
-		isResourceKinded: v.isResourceKinded,
-		dictionary:       dictionary,
-		isDestroyed:      v.isDestroyed,
-	}
+	dictionary := newDictionaryValueFromAtreeMap(
+		interpreter,
+		v.Type,
+		v.elementSize,
+		orderedMap,
+	)
```

Comment:
Calling `newDictionaryValueFromAtreeMap` here in `DictionaryValue.Clone()` is different from `CompositeValue.Clone()` which [creates `CompositeValue` directly](https://github.com/onflow/cadence/blob/b83b7db3aa03c0d72b031170b04bc995832cab64/runtime/interpreter/value.go#L16318-16333).

`newDictionaryValueFromAtreeMap` meters `DictionaryValueBaseMemoryUsage`.

Should `DictionaryValue.Clone()` and/or `CompositeValue.Clone()` be updated to meter more consistently?

Reply:
@fxamacker Actually, Clone is never metered, it's never used by the interpreter / during execution, only for e.g. migrations. I removed the unnecessary metering in [351c3c6](https://github.com/onflow/cadence/pull/2509/commits/351c3c6d678b2a8adf33562df29eeebc8298a1f8). Could you please have another look if that makes sense?

Reply:
Makes sense!  Thanks.